spyder/widgets/collectionseditor.py

```diff
@@ -926,6 +934,10 @@ def mouseDoubleClickEvent(self, event):
         else:
             event.accept()
 
+    def mouseMoveEvent(self, event):
+        """Change cursor shape."""
+        self.setCursor(Qt.PointingHandCursor)
```

Comment:
In order to change the cursor only when there is a row being hovered we need to change this for something like:

```suggestion
        if self.rowAt(event.y()) != -1:
            self.setCursor(Qt.PointingHandCursor)
        else:
            self.setCursor(Qt.ArrowCursor)
```